### PR TITLE
stage2: implement cImport

### PR DESCRIPTION
**Pull request by Vexu**
~I have no idea if any of this works since I can't build stage2 with LLVM.~ Still can't build stage2 with LLVM but `-fno-stage1` will work for now.

### CODE REVIEW

File: src/Sema.zig [hunk 1]
```diff
@@ -8082,7 +8150,15 @@ fn zirCDefine(
 ) CompileError!Air.Inst.Ref {
     const extra = sema.code.extraData(Zir.Inst.BinNode, extended.operand).data;
     const src: LazySrcLoc = .{ .node_offset = extra.node };
-    return sema.mod.fail(&block.base, src, "TODO: implement Sema.zirCDefine", .{});
+
+    const name = try sema.resolveConstString(block, src, extra.lhs);
+    if (sema.typeOf(extra.rhs).zigTypeTag() != .Void) {
+        const value = try sema.resolveConstString(block, src, extra.lhs);
```
Comment (g-w1) [inline src/Sema.zig:8156]:
```suggestion
        const value = try sema.resolveConstString(block, src, extra.rhs);
```

**Review by g-w1 — CHANGES_REQUESTED** [review]
If you want to build stage2 with llvm,
`make` or `ninja` should be sufficient to test this.

**Comment by g-w1** [comment]
cDefine("THING") does not work as it has 2 params here
https://github.com/ziglang/zig/blob/2a0c44fff3506cdc072d67374e7ffca495cebdc7/src/BuiltinFn.zig#L260-L264
It may need to be special cased in astgen

**Comment by Vexu** [comment]
> cDefine("THING") does not work as it has 2 params here


It doesn't work in stage1 either:
https://github.com/ziglang/zig/blob/2a0c44fff3506cdc072d67374e7ffca495cebdc7/src/stage1/codegen.cpp#L9047

**Comment by g-w1** [comment]
Oh, sorry I was mistaken, I read the code wrong :(

**Comment by andrewrk** [comment]
Got this error when running the behavior tests:

```
[nix-shell:~/dev/zig/build-release]$ ./zig-out/bin/zig test ../test/behavior.zig -fLLVM 
Segmentation fault at address 0x0
???:?:?: 0x7fa38740863d in ??? (???)
/home/andy/dev/zig/src/Module.zig:1586:17: 0x29315ae in Module.ErrorMsg.deinit (zig)
        gpa.free(err_msg.msg);
                ^
/home/andy/dev/zig/src/Module.zig:1583:24: 0x2931561 in Module.ErrorMsg.deinit (zig)
            note.deinit(gpa);
                       ^
/home/andy/dev/zig/src/Module.zig:1565:23: 0x2850cbc in Module.ErrorMsg.destroy (zig)
        err_msg.deinit(gpa);
                      ^
/home/andy/dev/zig/src/Module.zig:2331:22: 0x283d212 in Module.deinit (zig)
        value.destroy(gpa);
                     ^
/home/andy/dev/zig/src/Compilation.zig:1617:48: 0x2806b5b in Compilation.destroy (zig)
    if (optional_module) |module| module.deinit();
                                               ^
/home/andy/dev/zig/src/main.zig:2130:44: 0x2797732 in buildOutputType (zig)
    defer if (!comp_destroyed) comp.destroy();
                                           ^
/home/andy/dev/zig/src/main.zig:209:31: 0x2780286 in mainArgs (zig)
        return buildOutputType(gpa, arena, args, .zig_test);
                              ^
/home/andy/dev/zig/src/main.zig:158:20: 0x277f035 in main (zig)
    return mainArgs(gpa, arena, args);
                   ^
/home/andy/dev/zig/lib/std/start.zig:515:37: 0x2bd9a87 in std.start.callMain (zig)
            const result = root.main() catch |err| {
                                    ^
/home/andy/dev/zig/lib/std/start.zig:457:12: 0x2781ce7 in std.start.callMainWithArgs (zig)
    return @call(.{ .modifier = .always_inline }, callMain, .{});
           ^
/home/andy/dev/zig/lib/std/start.zig:422:12: 0x2781a92 in std.start.main (zig)
    return @call(.{ .modifier = .always_inline }, callMainWithArgs, .{ @intCast(usize, c_argc), c_argv, envp });
           ^
Aborted (core dumped)
```

I pushed a commit to the branch before merging to solve this (followed the pattern set by other messages with error notes).

**Comment by andrewrk** [comment]
great work! so many new test cases passing :star2: 

> Still can't build stage2 with LLVM

hmmmmm do we need to get the C backend up to all behavior tests passing so that you can build stage2 and test the frontend without llvm? :)

**Pull request merged**